pkg/controller/plan/validation.go

```diff
@@ -751,49 +788,98 @@ func (r *Reconciler) validateVddkImage(plan *api.Plan) (err error) {
 		Message:  "Validating VDDK init image",
 	}
 
-	source := plan.Referenced.Provider.Source
-	if source == nil {
-		return
+	if len(job.Status.Conditions) == 0 {
+		r.Log.Info("validation of VDDK job is in progress", "image", image)
+		plan.Status.SetCondition(vddkValidationInProgress)
 	}
-	destination := plan.Referenced.Provider.Destination
-	if destination == nil {
+	var ctx *plancontext.Context
+	ctx, err = plancontext.New(r, plan, r.Log)
+	if err != nil {
 		return
 	}
-
-	if source.Type() != api.VSphere {
-		// VDDK is not used for other provider types
+	pods := &core.PodList{}
+	if err = ctx.Destination.Client.List(context.TODO(), pods, &client.ListOptions{
+		Namespace:     plan.Spec.TargetNamespace,
+		LabelSelector: k8slabels.SelectorFromSet(map[string]string{"job-name": job.Name}),
+	}); err != nil {
 		return
 	}
+	for _, pod := range pods.Items {
+		if strings.Contains(pod.Name, job.Name) {
+			jobCopy := job.DeepCopy()
+			// there should only be a single init container
+			initStatus := pod.Status.InitContainerStatuses[0]
+			if initStatus.State.Waiting != nil {
+				r.Log.Info("Waiting for init container", "job", job.Name, "state", initStatus)
+				cnd := libcnd.Condition{
+					Type:     VDDKInitImageNotReady,
+					Status:   True,
+					Reason:   initStatus.State.Waiting.Reason,
+					Category: Advisory,
+					Message:  initStatus.State.Waiting.Message,
+					Durable:  false,
+				}
+				plan.Status.SetCondition(cnd)
 
-	if image, found := source.Spec.Settings[api.VDDK]; found {
-		var job *batchv1.Job
-		if job, err = r.ensureVddkImageValidationJob(plan, image); err != nil {
-			return
-		}
-		if len(job.Status.Conditions) == 0 {
-			r.Log.Info("validation of VDDK job is in progress", "image", image)
-			plan.Status.SetCondition(vddkValidationInProgress)
+				// If a job fails due to exceeding the deadline, the pods will be deleted and
+				// we won't be able to determine what caused the failure. So persist any
+				// initStatus errors via job annotations so that they we can use them later.
```

Comment:
So we will be constatnly failing until we get the image?
What is the difference between this and removing the `activeDeadlineSeconds`?

Reply:
Do we know if we start another pull when the job restarts?
If we will restart the job multiple times we might start more pulls